src/components/MainSection.jsx

```diff
@@ -1,111 +1,117 @@
+          onClick={() => setFilterCategory('Dinner')}
+        >
+          Dinner
+        </button>
+        <button
+          className={`menu-button ${filterCategory === 'Breakfast' ? 'active' : ''}`}
+          onClick={() => setFilterCategory('Breakfast')}
+        >
+          Breakfast
+        </button>
+        <button
+          className={`menu-button ${filterCategory === 'Dessert' ? 'active' : ''}`}
+          onClick={() => setFilterCategory('Dessert')}
+        >
+          Dessert
+        </button>
+      </div>
 
-        <div className="menu">
-          {itemsToDisplay.map((item) => {
-            if (!this.quantityRefs[item.id]) {
-              this.quantityRefs[item.id] = createRef();
-            }
-            return (
-              <div key={item.id} className="menu-item">
-                <img src={item.img} alt={item.meal} className="menu-item-image" />
-                <div className="menu-item-content">
-                  <h2 className="menu-item-name">{item.meal}</h2>
-                  <p className="menu-item-price">${item.price.toFixed(2)} USD</p>
-                  <div className="item-actions">
-                    <input
-                      type="number"
-                      min="1"
-                      defaultValue="1"
-                      className="quantity-input"
-                      ref={this.quantityRefs[item.id]}
-                    />
-                    <button
-                      className="menu-button"
-                      onClick={() => this.handleAddToCart(item.id)}
-                    >
-                      Add to cart
-                    </button>
-                  </div>
+      <div className="menu">
+        {itemsToDisplay.map((item) => {
+          if (!quantityRefs.current[item.id]) {
+            quantityRefs.current[item.id] = React.createRef();
+          }
+          return (
+            <div key={item.id} className="menu-item">
+              <img src={item.img} alt={item.meal} className="menu-item-image" />
+              <div className="menu-item-content">
+                <h2 className="menu-item-name">{item.meal}</h2>
+                <p className="menu-item-price">${item.price.toFixed(2)} USD</p>
+                <div className="item-actions">
+                  <input
+                    type="number"
+                    min="1"
+                    defaultValue="1"
+                    className="quantity-input"
+                    ref={(el) => (quantityRefs.current[item.id] = el)}
+                    onChange={() => {}}
+                  />
+                  <button
+                    className="menu-button"
+                    onClick={() => handleAddToCart(item.id)}
+                    disabled={
+                      !quantityRefs.current[item.id] ||
+                      !quantityRefs.current[item.id].value ||
+                      Number(quantityRefs.current[item.id].value) <= 0
```

Comment:
Such long statements better to put to a constant. For example:
const isMenuBtnDisabled =  !quantityRefs.current[item.id] || !quantityRefs.current[item.id].value || Number(quantityRefs.current[item.id].value <= 0